### PR TITLE
Feature/5

### PR DESCRIPTION
**Pull request by kiber**
DEV-44834

### CODE REVIEW

File: Api/BlockManagerInterface.php [hunk 1]
```diff
@@ -6,14 +6,14 @@
 {
     /**
      * @param int $blockId
-     * @return \Magento\Cms\Api\Data\BlockInterface
+     * @return \Snowdog\CmsApi\Api\Data\BlockInterface
```
Comment (mcjwsk) [inline Api/BlockManagerInterface.php]: You have changed the return type, but the implementation of this interface still returns an instance of `\Magento\Cms\Api\Data\BlockInterface`.

File: Api/BlockManagerInterface.php [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
      */
     public function getById($blockId);
```
Comment (mcjwsk) [inline Api/BlockManagerInterface.php]: It is requested (in #5) to add stores to the result of `getList` method only. Adding this to `getById` and `getByIdentifier` isn't necessary now.

File: Model/Block.php [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+/**
+ * CMS block model
+ *
+ * @method ResourceCmsBlock _getResource()
+ * @method ResourceCmsBlock getResource()
+ * @method Block setStoreId(array $storeId)
+ * @method array getStoreId()
+ */
+class Block extends AbstractModel implements BlockInterface, IdentityInterface
```
Comment (mcjwsk) [inline Model/Block.php]: Shouldn't it just extend from `\Magento\Cms\Model\Block` (so you add `getStores()` and `setStores()` methods only)?

File: Api/Data/BlockInterface.php [hunk 4]
```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Snowdog\CmsApi\Api\Data;
+
+/**
+ * CMS block interface.
+ * @api
+ */
+interface BlockInterface
```
Comment (mcjwsk) [inline Api/Data/BlockInterface.php]: Can't it extend from `\Magento\Cms\Api\Data\BlockInterface`?

**Review by mcjwsk — CHANGES_REQUESTED** [review]
You forgot about cms pages.

**Review by mcjwsk — APPROVED** [review]

**Review by jochhop — APPROVED** [review]

**Pull request merged**